src/main/java/net/consensys/orion/exception/OrionException.java

```diff
@@ -40,4 +42,21 @@ public OrionException(final OrionErrorCode code, final String message, @Nullable
   public OrionErrorCode code() {
     return code;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    OrionException that = (OrionException) o;
+    return code == that.code;
```

Comment:
nit: This is perfectly valid and correct (as both are enums) - does using
code.equals(that.code) make less future effort? Other option was Objects.equals(code, that.code).
Happy to leave it to your discretion

Reply:
I don't mind changing it to Objects.equal